### PR TITLE
chore(eslint): update rules - FRONT-1247

### PR DESCRIPTION
**Pull request by kalinchernev**
I've added a documentation page with the rules I've been following https://github.com/ec-europa/europa-component-library/compare/chore/lint-rules-FRONT-1247?expand=1#diff-babc1190f732dbdad29033d18652a6e7

### CODE REVIEW

File: docs/conventions/linting.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+- local override for the website at `src/website/.eslintrc.js`
+
+There are few more overrides which are less frequently necessary for the overall coding flow of the project. Keep in mind that the closer a configuration file is to a given file, an overriding takes place. Thus, when updating rules, pay attention to scope of rules.
+
+## Specifics
+
+Here are a few useful details to keep in mind when maintaining the set of linting rules for the ECL project:
+
+- `import/no-extraneous-dependencies` is set throughout because of the lerna workspace features
+- `no-param-reassign` is
```
Comment (emeryro) [inline docs/conventions/linting.md]: unwanted new line I guess

File: src/systems/ec/implementations/react/components/description-list/src/DescriptionListWithData.jsx [hunk 2]
```diff
@@ -32,10 +32,10 @@ const DescriptionListWithData = ({ items, className, ...props }) => {
         }
 
         return (
-          <Fragment key={item.term}>
+          <div key={item.term}>
```
Comment (emeryro) [inline src/systems/ec/implementations/react/components/description-list/src/DescriptionListWithData.jsx]: adding extra div item here breaks the design.
the `key` property is there only for React, so I'm not sure it is an issue to have it on a Fragment (this seems to be a standard way to handle this https://medium.com/@yurachoi/react-js-fragments-175290da6435)

File: src/systems/ec/implementations/react/components/timeline2/src/Timeline2.jsx [hunk 3]
```diff
@@ -47,7 +46,7 @@ export const Timeline2 = ({
       {items &&
         items.map(
           ({ id: itemId, content: itemContent, ...itemProps }, index) => (
-            <Fragment key={itemId}>
+            <div key={itemId}>
```
Comment (emeryro) [inline src/systems/ec/implementations/react/components/timeline2/src/Timeline2.jsx]: same remark here, the extra div breaks the display

File: src/systems/ec/implementations/react/deprecated/timeline/src/Timeline.jsx [hunk 4]
```diff
@@ -21,7 +20,7 @@ export const Timeline = ({ items, button, className, ...props }) => {
   return (
     <ol {...props} className={classNames} data-ecl-timeline>
       {items.map((item, index) => (
-        <Fragment key={item.id}>
+        <div key={item.id}>
```
Comment (emeryro) [inline src/systems/ec/implementations/react/deprecated/timeline/src/Timeline.jsx]: same here

**Review by emeryro — CHANGES_REQUESTED** [review]
having these extra div instead of Fragment would be an issue (for both EC and EU).
See my comments below

File: src/systems/ec/implementations/react/components/description-list/src/DescriptionListWithData.jsx (hunk 2 above)
Comment (kalinchernev) [reply src/systems/ec/implementations/react/components/description-list/src/DescriptionListWithData.jsx]: I used the resource provided and put back the `React.Fragment` version which does not trigger linting rules.

File: src/systems/eu/implementations/react/components/description-list/src/DescriptionListWithData.jsx [hunk 5]
```diff
@@ -32,10 +32,10 @@ const DescriptionListWithData = ({ items, className, ...props }) => {
         }
 
         return (
-          <Fragment key={item.term}>
+          <div key={item.term}>
```
Comment (emeryro) [inline src/systems/eu/implementations/react/components/description-list/src/DescriptionListWithData.jsx]: There are still a few extra div to be converted to Fragment, here

File: src/systems/eu/implementations/react/components/timeline/src/Timeline.jsx [hunk 6]
```diff
@@ -21,7 +20,7 @@ export const Timeline = ({ items, button, className, ...props }) => {
   return (
     <ol {...props} className={classNames} data-ecl-timeline>
       {items.map((item, index) => (
-        <Fragment key={item.id}>
+        <div key={item.id}>
```
Comment (emeryro) [inline src/systems/eu/implementations/react/components/timeline/src/Timeline.jsx]: here

File: src/systems/eu/implementations/react/components/timeline2/src/Timeline2.jsx [hunk 7]
```diff
@@ -47,7 +46,7 @@ export const Timeline2 = ({
       {items &&
         items.map(
           ({ id: itemId, content: itemContent, ...itemProps }, index) => (
-            <Fragment key={itemId}>
+            <div key={itemId}>
```
Comment (emeryro) [inline src/systems/eu/implementations/react/components/timeline2/src/Timeline2.jsx]: and here

**Review by emeryro — CHANGES_REQUESTED** [review]

File: src/systems/eu/implementations/react/components/description-list/src/DescriptionListWithData.jsx (hunk 5 above)
Comment (kalinchernev) [reply src/systems/eu/implementations/react/components/description-list/src/DescriptionListWithData.jsx]: Sorry, updated these as well

**Review by emeryro — APPROVED** [review]

**Pull request merged**